aptos-move/aptos-vm/src/keyless_validation.rs

```diff
@@ -28,6 +30,10 @@ macro_rules! value_deserialization_error {
     }};
 }
 
+/// TODO(keyless): Comments say Cache should be wrapped in an Arc if used from multiple threads, but do I even need one if it's in a sync::Lazy?
+pub(crate) static ZKP_CACHE: Lazy<Cache<Groth16ProofAndStatement, bool>> =
+    Lazy::new(|| Cache::new(1_000_000));
```

Comment:
in case vk is rotated, we will need to empty the cache?

Reply:
is there timeout for cached entry, or they will be there forever unless evicted?

Reply:
at least the vk needs to be part of the cache key, otherwise execution result may diverge?

Reply:
That's a great point! Maybe I can keep track of the last seen VK from on-chain & compare it to the current one. If different, clear the cache.